### ISSUE TITLE
Impossible condition match

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#459]
It should works similar to Java version. If you block your code by some conditional code, maybe you have some impossible branches. For instance:

```php
if ($user === null) {
     return null;
}

if ($user === null) {
    $nullUsers++;
}
```

The second _if()_ is impossible, once that the first one have the same condition but stop the code execution because of the `return`.

Same should be valid for _else_ conditions, like:

```php
if ($user === null) {
     return null;
}

if ($user !== null) {
    $nullUsers++; // Ok now, because of !==
}
else {
    $notNullUsers++; // Not ok
}
```

**Comment by funivan** [comment]
Hello. I try to investigate variable types and detect similar cases. This is a really hard task.  If i find solution I will ping you back @rentalhost.

**Comment by kalessil** [comment]
Currently PhpStan does report the case, hence Iet's drop this ticket from our scope. It also a complex one, I'd like to focus on our primary scope.
